### DISCUSSION TITLE
Please help: What is the reason for the failure of lighthouse geometry definition?

### DISCUSSION THREAD

**Discussion by Zm442709994** [bitcraze/discussions#668] [Q&A]
Followed the instructions in the documentation to complete the geometry definition, but encountered a failure in the last step and was prompted to redefine. I have tried several times, and the geometry definition was successful before today, but it cannot be defined successfully today.
![7f4ecdd6-a002-4571-812e-7a3a7b7c39dc](https://user-images.githubusercontent.com/118723077/233026216-3f4c9355-079a-4f95-9099-0e4343420964.jpeg)

**Comment by krichardsson** [comment]
Unfortunately it is very hard to say what the problem might be. 

> but encountered a failure in the last step

The estimation process works by first collecting data through the first part of the wizard, and analyzing it only in the last step. This is why it is not found until the last step.  

Since you have managed to estimate the geometry earlier in the same system, I think the question is if anything has changed in the system? 
Some possible reasons could be:
* reflections
* direct sunlight
* other IR-light sources (like a mocap system)
* Have the channels of the base stations been re-configured?

**Reply by Zm442709994** [reply]
There is a new issue: The unmanned aerial vehicle can connect normally when it is outside the range of the lighthouse base station.  However, once it enters the infrared light range of the lighthouse base station, the UAV will restart and disconnect from the client.  It cannot be connected again unless it leaves the range of the lighthouse base station and restarts.

**Reply by krichardsson** [reply]
This is probably because the calibration data has is corrupt (a bug fixed in https://github.com/bitcraze/crazyflie-firmware/pull/1266), this would perhaps also explain why the geometry estimation does not work.

You can find a way to fix the problem in #541

**Reply by Zm442709994** [reply]
Thank you very much for providing the solution to the drone restart issue. Below is the detailed operational process that I previously mentioned, which cannot define geometry. Could you please take a look at any possible issues? Many thanks.
https://youtu.be/ml6Sswt7dv0
console
![6](https://user-images.githubusercontent.com/118723077/233559922-93382ea2-ae70-44be-9f8a-3318f26c6968.png)

**Reply by faezehmhmdi** [reply]
@Zm442709994 @krichardsson Did you manage to solve this issue? I also used to calibrate the drone with lighthouse position before, but now I receive "ESTKALMAN: State out of bounds, resetting" in the Console tab and cfclient closes suddenly when it wants to estimate the geometry on the last calibration state.

**Reply by krichardsson** [reply]
@faezehmhmdi This sounds like a slightly different problem, please start a new discussion thread for it

**Comment by krichardsson** [comment]
The line `LH: Warning: not getting data from all sensors` indicates that one of the sensors on your lighthouse deck does not work properly. All four sensors are required for the geometry estimation to work. 
Can you see any visible damage to any sensor, or do you have another deck to try?

**Reply by Zm442709994** [reply]
There is only one deck now, and no obvious issues have been found on the surface.
![正面](https://user-images.githubusercontent.com/118723077/233581311-a1d49c0f-3d87-46ed-987f-d48596b82662.jpg)
![反面](https://user-images.githubusercontent.com/118723077/233581304-ec8c161a-76d6-43b3-83d9-d1087dad80db.jpg)

**Reply by Zm442709994** [reply]
I don't see any obvious issues, but it's clear that sensor #2 is having problems. Can you tell if there's a faulty component and is there any way to confirm which part is malfunctioning?

**Reply by krichardsson** [reply]
Indeed, nothing is obviously broken, please send us an email on support@bitcraze.io, we'll handle it there.
Also include information on when/where you bought it, order number and so on.
